core/consensus/slots/src/slots.rs

```diff
@@ -23,29 +23,32 @@ use consensus_common::Error;
 use futures::prelude::*;
 use futures::try_ready;
 use inherents::{InherentData, InherentDataProviders};
-use log::warn;
+
 use std::marker::PhantomData;
 use std::time::{Duration, Instant};
 use tokio_timer::Delay;
 
 /// Returns current duration since unix epoch.
-pub fn duration_now() -> Option<Duration> {
+pub fn duration_now() -> Duration {
 	use std::time::SystemTime;
 
 	let now = SystemTime::now();
-	now.duration_since(SystemTime::UNIX_EPOCH)
-		.map_err(|e| {
-			warn!(
-				"Current time {:?} is before unix epoch. Something is wrong: {:?}",
-				now, e
-			);
-		})
-		.ok()
+	match now.duration_since(SystemTime::UNIX_EPOCH) {
+		Ok(s) => s,
+		Err(e) => panic!(
+			"Current time {:?} is before unix epoch. Something is wrong: {:?}",
+			now, e
+		),
+	}
 }
 
 /// Get the slot for now.
-pub fn slot_now(slot_duration: u64) -> Option<u64> {
-	duration_now().map(|s| s.as_secs() / slot_duration)
+pub fn slot_now(slot_duration: u64, offset: Duration, is_positive: bool) -> u64 {
```

Comment:
Will the offset be used in the future?

Reply:
Yes, it will ― it is the difference between local time and on-chain time.

Reply:
I agree, but I also think that can be a later refactoring.

Reply:
I really would like to see it here, this is no big refactoring and moving something to the future means it will probably never done ;)